app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java

```diff
@@ -446,23 +445,62 @@ public static String relativePath(File src, File dest) {
 
         String t;
         try {
-            if (!TextViewUtils.isNullOrWhitespace(t = Files.probeContentType(file.toPath()))) {
+            if (!GsTextUtils.isNullOrEmpty(t = Files.probeContentType(file.toPath()))) {
                 return t;
             }
         } catch (Exception ignored) {
         }
 
         if (file.exists() && file.isFile()) {
             try (InputStream is = new BufferedInputStream(new FileInputStream(file))) {
-                if (!TextViewUtils.isNullOrWhitespace(t = Files.probeContentType(file.toPath()))) {
+                if (!GsTextUtils.isNullOrEmpty(t = Files.probeContentType(file.toPath()))) {
                     return t;
                 }
             } catch (Exception ignored) {
             }
         }
 
-        t = URLConnection.guessContentTypeFromName(file.getName().replace(".jenc", ""));
-        return TextViewUtils.isNullOrWhitespace(t) ? "*/*" : t;
+        if (!GsTextUtils.isNullOrEmpty(t = URLConnection.guessContentTypeFromName(file.getName().replace(".jenc", "")))) {
+            return t;
+        }
+
+        // Try extracting by running shell file -b on the file - for textfiles this very often results in "ASCII text"
+        try {
+            final String cmd = "file -b " + file.getAbsolutePath().replace(" ", "\\ ").replace(";", "").replace("&", "").replace("$", "");
+            final Process p = Runtime.getRuntime().exec(cmd);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+            t = reader.readLine().trim().toLowerCase(Locale.ROOT);
```

Comment:
@harshad1 

I made the changes on master and merged it into your branch. I'm done for now.

What is probably the newest best addition is this section here. As even with all the many things Android & Java has to other, it still is by ~10 means somehow incapable to detect files if they are not uber-popular like .jpg or .txt is. 

What I found is that the normal shell `file` utility however works quite well for many things, where the java & android utils fail. So for many plaintext files it just echoes "ASCII text", or  "language script" (like python script).

Now utilizing this info at Markor as well and so hopefully textfiles are way better detected now. I.e. tried it on ".taskpaper" which is also a plain textfile & format like todo.txt is -- everything fails to detect this, but this new one works!

Reply:
Very cool. However, we are relying on the `file` shipped with whatever version of android. IMO at this stage we need to consider shipping file or including a library

Reply:
Thats right, but I rather not include some C++/NDKK prebuilt native stuff on my own here 😄 . Lets see for a while how it works out. Looks to me promising that it works well / better then the others (for non-popular filetypes) so far.

Reply:
https://github.com/overview/mime-types should be investigated imo. We have certainly spent a lot of time on mime types

Reply:
it doesn't have a mapping for lets say asciidoc. so even when, we would need some manual mapping still. Just like for `.md.txt` or `.jenc` stuff.

![grafik](https://user-images.githubusercontent.com/6735650/192898574-abba9a2d-20ab-4cfc-9dc0-f4382cffe69b.png)

At the end, the interesting info for Markor is: Is it one of our ~4 "well supported" formats like Markdown, is it some other textfile (=plaintext format), or is it not text and thus some binary. And to current feeling that new file stat call seems to fill that textfile detection gap better then the rest so far :smiley: .